libcontainerd/client_windows.go

```diff
@@ -427,12 +427,80 @@ func (clnt *client) Resize(containerID, processFriendlyName string, width, heigh
 
 // Pause handles pause requests for containers
 func (clnt *client) Pause(containerID string) error {
-	return errors.New("Windows: Containers cannot be paused")
+	unlockContainer := true
+	// Get the libcontainerd container object
+	clnt.lock(containerID)
+	defer func() {
+		if unlockContainer {
+			clnt.unlock(containerID)
+		}
+	}()
+	container, err := clnt.getContainer(containerID)
+	if err != nil {
+		return err
+	}
+
+	for _, option := range container.options {
+		if h, ok := option.(*HyperVIsolationOption); ok {
+			if !h.IsHyperV {
+				return errors.New("cannot pause Windows Server Containers, use Hyper-V containers to pause on Windows")
```

Comment:
I think "cannot pause Windows Server Containers" both shortens the error, and starts it with a lowercase so that any tools checking error casing don't get tripped up by the proper noun Windows starting the error. WDYT?